lib/api/completeMultipartUpload.js

```diff
@@ -153,235 +154,79 @@ function completeMultipartUpload(authInfo, request, log, callback) {
-                            // part and the size the of the piece is the full
-                            // part size.
-                            const location = storedPart.locations[j];
-                            // If there is no location, move on
-                            if (!location || typeof location !== 'object') {
-                                continue;
-                            }
-                            let pieceSize = partSize;
-                            if (location.size) {
-                                pieceSize = Number.parseInt(location.size, 10);
-                            }
-                            const pieceRetrievalInfo = {
-                                key: location.key,
-                                size: pieceSize,
-                                start: calculatedSize,
-                                dataStoreName: location.dataStoreName,
-                                dataStoreETag,
-                                cryptoScheme: location.sseCryptoScheme,
-                                cipheredDataKey: location.sseCipheredDataKey,
-                            };
-                            dataLocations.push(pieceRetrievalInfo);
-                            calculatedSize += pieceSize;
-                        }
-
-                        const partETagWithoutQuotes =
-                            storedPart.ETag.slice(1, -1);
-                        concatETags += partETagWithoutQuotes;
-                        storedPartsAsObjects =
-                          storedPartsAsObjects.splice(1);
-                    } else {
-                        extraParts.push(storedPart);
-                        storedPartsAsObjects =
-                          storedPartsAsObjects.splice(1);
-                    }
-                }
-                if (!isPartUploaded) {
-                    return next(errors.InvalidPart, destBucket);
-                }
-            }
-            extraParts = extraParts.concat(storedPartsAsObjects);
-            // if extra parts, need to delete the data when done with completing
-            // mpu so extract the info to delete here
-            if (extraParts.length > 0) {
-                extraParts.forEach(part => {
-                    const locations = part.locations;
-                    locations.forEach(location => {
-                        if (!location || typeof location !== 'object') {
-                            return;
-                        }
-                        extraPartLocations.push(location);
-                    });
-                });
+            if (completeObjData && completeObjData.calculateEtag) {
+                const dataLocations = [
+                    {
+                        key: completeObjData.key,
+                        size: calculatedSize,
+                        start: 0,
+                        dataStoreName: storedMetadata.dataStoreName,
```

Comment:
Where do you get `storedMetadata.dataStoreName` from?

Reply:
It's stored in the metadata